lib/initial-setup/WP_Auth0_InitialSetup_Consent.php

```diff
@@ -170,9 +170,12 @@ public function consent_callback( $name ) {
 		if ( $should_create_and_update_connection ) {
 
 			if ( $connection_exists === false ) {
-				$migration_token = JWT::urlsafeB64Encode( openssl_random_pseudo_bytes( 64 ) );
-				$operations      = new WP_Auth0_Api_Operations( $this->a0_options );
-				$response        = $operations->create_wordpress_connection(
+				$migration_token = $this->a0_options->get( 'migration_token' );
+				if ( empty( $migration_token ) ) {
+					$migration_token = JWT::urlsafeB64Encode( openssl_random_pseudo_bytes( 64 ) );
```

Comment:
what if this "is empty logic" is part of the `$this->a0_options->get( 'migration_token' );`? would it make sense, can that be done? so that whenever you request it you always obtain a valid value.

Reply:
We don't really want that to happen. The value needs to be added to a script JS file and added to the Connection so generating one out of the blue would not be helpful. Also, there are times when we need to know if the setting is empty or not. Otherwise, we'd have to add an `has()` method or similar. These tokens are not generated often enough to make that worth it.